test/parallel/test-tls-set-sigalgs.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+const common = require('../common');
+if (!common.hasCrypto) common.skip('missing crypto');
+const fixtures = require('../common/fixtures');
+
+// Test cipher: option for TLS.
+
+const {
+  assert, connect, keys
+} = require(fixtures.path('tls-connect'));
+
+// There are shared algorithms -> successful connection
+connect({
+  client: {
+    checkServerIdentity: (servername, cert) => { },
+    ca: `${keys.agent1.cert}\n${keys.agent6.ca}`,
+    cert: keys.agent2.cert,
+    key: keys.agent2.key,
+    sigalgs: 'RSA-PSS+SHA384',
```

Comment:
I suggest adding one more test, with multiple algs, showing that one of the algs is chosen if it matches the peer.

Reply:
This would be nice, but I can't really imagine how it could be done. There is an API call to get currently used cipher (`SSL_get_current_cipher`), but nothing I could find to introspect what signature algorithm was used for the handshake.

Reply:
That would be a useful addition in its own right and should arguably be part of this PR (but it could be a separate commit.)

The OpenSSL APIs to use are `SSL_get_sigalgs()` and `SSL_get_shared_sigalgs()`. The first one returns the list of signature algorithms that the peer supports, the second one is the intersection of the peer's list and ours.

https://www.openssl.org/docs/man1.1.1/man3/SSL_get_shared_sigalgs.html